scripts/collection_from_items.py

```diff
@@ -6,19 +6,92 @@
 from boto3 import client
 from linz_logger import get_log
 
-from scripts.cli.cli_helper import coalesce_multi_single
+from scripts.cli.cli_helper import coalesce_multi_single, nullable_str, valid_date
 from scripts.files.fs_s3 import bucket_name_from_path, get_object_parallel_multithreading, list_json_in_uri
 from scripts.logging.time_helper import time_in_ms
 from scripts.stac.imagery.collection import ImageryCollection
+from scripts.stac.imagery.generate_metadata import ElevationSubtypes, ImagerySubtypes, generate_description, generate_title
 from scripts.stac.imagery.provider import Provider, ProviderRole
 
 
+# pylint: disable-msg=too-many-locals
 def main() -> None:
     parser = argparse.ArgumentParser()
+    parser.add_argument("--preset", dest="preset", required=True, help="Standardised file format. Example: webp")
     parser.add_argument("--uri", dest="uri", help="s3 path to items and collection.json write location", required=True)
     parser.add_argument("--collection-id", dest="collection_id", help="Collection ID", required=True)
-    parser.add_argument("--title", dest="title", help="Collection title", required=True)
-    parser.add_argument("--description", dest="description", help="Collection description", required=True)
+    parser.add_argument(
+        "--subtype",
+        dest="subtype",
+        help="Dataset subtype description",
+        required=True,
+        choices=[
+            ImagerySubtypes.AERIAL,
+            ImagerySubtypes.HISTORICAL,
+            ImagerySubtypes.SATELLIE,
+            ImagerySubtypes.URBAN,
+            ImagerySubtypes.RURAL,
+            ElevationSubtypes.DEM,
+            ElevationSubtypes.DSM,
+        ],
+    )
+    parser.add_argument(
+        "--region",
+        dest="region",
+        help="Region of Dataset",
+        required=True,
+        choices=[
+            "antarctica",
+            "auckland",
+            "bay-of-plenty",
+            "canterbury",
+            "gisborne",
+            "global",
+            "hawkes-bay",
+            "manawatu-whanganui",
+            "marlborough",
+            "nelson",
+            "new-zealand",
+            "northland",
+            "otago",
+            "pacific-islands",
+            "southland",
+            "taranaki",
+            "tasman",
+            "waikato",
+            "wellington",
+            "west-coast",
+        ],
+    )
+    parser.add_argument("--gsd", dest="gsd", help="GSD of imagery Dataset", type=nullable_str, required=True)
```

Comment:
Is it really valid to specify `--gsd=''`, so that `args.gsd` ends up being `None`? If so, why not just make this optional?

Reply:
gsd is required for the title so, I have changed (locally atm) the workflow parameter to:
```
- name: gsd
description: 'Dataset GSD (e.g., 0.3m)'
```
If the user tries to submit an imagery-standardise workflow without specifying a gsd parameter it will fail. 
Therefore I have made args.gsd here a str